src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -273,18 +335,917 @@ private static IEnumerable<SyntaxNode> GetOwningNodes(SyntaxNode root, int posit
+
+            var newNodeAfterInsertion = newRoot.GetAnnotatedNodes(s_replacementNodeAnnotation).Single();
+            var formattingNode = newNodeAfterInsertion.FirstAncestorOrSelf<MemberDeclarationSyntax>() ?? newNodeAfterInsertion;
+            var formattedNewRoot = Formatter.Format(
+                newRoot,
+                formattingNode.Span,
+                document.Project.Solution.Workspace,
+                cancellationToken: cancellationToken);
+            var nextCaretPosition = formattedNewRoot.GetAnnotatedTokens(s_openBracePositionAnnotation).Single().Span.End;
+            return (formattedNewRoot, nextCaretPosition);
+        }
+
+        private static SyntaxNode ReplaceNodeAndFormat(
+            Document document,
+            SyntaxNode root,
+            SyntaxNode oldNode,
+            SyntaxNode newNode,
+            CancellationToken cancellationToken)
+        {
+            var annotatedNewNode = newNode.WithAdditionalAnnotations(s_replacementNodeAnnotation);
+            var newRoot = root.ReplaceNode(
+                oldNode,
+                annotatedNewNode);
+            var newNodeAfterInsertion = newRoot.GetAnnotatedNodes(s_replacementNodeAnnotation).Single();
+
+            var formattingNode = newNodeAfterInsertion.FirstAncestorOrSelf<MemberDeclarationSyntax>() ?? newNodeAfterInsertion;
+            var options = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var formattedNewRoot = Formatter.Format(
+                newRoot,
+                formattingNode.Span,
+                document.Project.Solution.Workspace,
+                options,
+                cancellationToken: cancellationToken);
+            return formattedNewRoot;
+        }
+
+        private static ObjectCreationExpressionSyntax WithArgumentListIfNeeded(ObjectCreationExpressionSyntax objectCreationExpressionNode)
+        {
+            var argumentList = objectCreationExpressionNode.ArgumentList;
+            var hasArgumentList = argumentList != null && !argumentList.IsMissing;
+            if (!hasArgumentList)
+            {
+                // Make sure the trailing trivia is passed to the argument list
+                // like var l = new List\r\n =>
+                // var l = new List()\r\r
+                var typeNode = objectCreationExpressionNode.Type;
+                var newArgumentList = SyntaxFactory.ArgumentList().WithTriviaFrom(typeNode);
+                var newTypeNode = typeNode.WithoutTrivia();
+                return objectCreationExpressionNode.WithType(newTypeNode).WithArgumentList(newArgumentList);
+            }
+
+            return objectCreationExpressionNode;
+        }
+
+        /// <summary>
+        /// Remove the brace for the input syntax node
+        /// For ObjectCreationExpressionSyntax, it would remove the initializer
+        /// For PropertyDeclarationSyntax, it would change it to a FieldDeclaration
```

Comment:
this is an interesting case.  why do we do this.  when does the user want to move from prop to field?  could this be frustrating?

Reply:
Sam and I have discussed about those scenarios and we think this would be a useful case.
If use invokes the command for a field with no semicolon, then we think they are trying to type an property, so in such case, the field is changed to a property.
If the command is invoked again in the context of property, then it means user doesn't want a property. Then the command remove the parenthesis and append a semicolon for the user.
tag @sharwell  to share more thoughts.